src/firmware/app.c

```diff
@@ -232,23 +232,25 @@ void app_set_assist_level(uint8_t level)
 	}
 }
 
+static __code uint8_t assist_mode_select_[] = {
```

Comment:
Use static const instead of static __code. I have verified SDCC will but const in data segment. Also please move this table inside the app_set_lights function right at the top.